File: ePreschool.Core/Entities/MonthlyPayment.cs
using ePreschool.Core.Entities.Base;
using ePreschool.Core.Enumerations;

namespace ePreschool.Core.Entities
{
    public class MonthlyPayment : BaseEntity
    {
        public bool IsPaid { get; set; }
        public int Month { get; set; }
        public int Year { get; set; }
        public int ParentId { get; set; }
        public Parent Parent { get; set; }
        public int ChildId { get; set; }
        public Child Child { get; set; }
        public decimal Price { get; set; }
        public float? Discount { get; set; }
        public string? Note { get; set; }
        public StatusPayment? Status { get; set; }

    }
}

File: UI/epreschool_admin/lib/models/monthlyPayment.dart

import 'package:epreschool_admin/models/parent.dart';

import 'base.dart';
import 'child.dart';

class MonthlyPayment extends BaseModel {
  late bool isPaid;
  late int month;
  late int year;
  late int parentId;
  late Parent? parent;
  late int childId;
  late Child? child;
  late int price;
  late int discount;
  late String note;
  late int status;

MonthlyPayment();

MonthlyPayment.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    isPaid = json['isPaid'];
    month=json['month'];
    year=json['year'];
    parentId = json['parentId'];
    if (json['parent'] != null) {
      parent = Parent.fromJson(json['parent']);
    }
    childId = json['childId'];
    if (json['child'] != null) {
      child = Child.fromJson(json['child']);
    }
    price = json['price'];
    discount = json['discount'];
    note = json['note'] ?? '';
    status = json['status'] ?? 0;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['isPaid'] = isPaid;
    data['month'] = month;
    data['year'] = year;
    data['parentId'] = parentId;
    data['childId'] = childId;
    data['price'] = price;
    data['discount'] = discount;
    data['note'] = note;
    data['status'] = status;
    return data;
  }
}